api/test/integration/test_agents_endpoints.tavern.yaml

```diff
@@ -0,0 +1,5044 @@
+          older_than: 20m
+          total_affected_items: 0
+          total_failed_items: 1
+        message: !anystr
+
+  - name: Try remove agent by name
+
+    request:
+      url: "{protocol:s}://{host:s}:{port:d}/agents"
+      method: DELETE
+      headers:
+        Authorization: "Bearer {test_login_token}"
+      params:
+        list_agents: bad_id
+
+    response:
+      status_code: 400
+      body: 
+        <<: *error_spec
+
+  - name: Remove agents
+
+    request:
+      url: "{protocol:s}://{host:s}:{port:d}/agents"
+      method: DELETE
+      headers:
+        Authorization: "Bearer {test_login_token}"
+      params:
+        list_agents: 004,005
+        older_than: '1s'
+        purge: True
+
+    response:
+      status_code: 200
+      body:
+        data:
+          affected_items: ["004","005"]
+          older_than: 1s
+          total_affected_items: 2
+        message: !anystr
+
+---
+test_name: DELETE /agents/:agent_id
+
+marks:
+  - usefixtures:
+      - agents_test
+
+stages:
+  - name: Try remove agent 000 (manager)
+
+    request:
+      url: "{protocol:s}://{host:s}:{port:d}/agents/000"
+      method: DELETE
+      headers:
+        Authorization: "Bearer {test_login_token}"
+
+    response: &error_delete_agent
+      status_code: 200
+      body:
+        data:
+          affected_items: []
+          failed_items:
+            - error:
+                code: 1703
+                message: !anystr
+                remediation: !anystr
+              id: '000'
+        message: !anystr
+
+  - name: Try remove not existing agent
+
+    request:
+      url: "{protocol:s}://{host:s}:{port:d}/agents/999"
+      method: DELETE
+      headers:
+        Authorization: "Bearer {test_login_token}"
+
+    response:
+      <<: *error_delete_agent
+      body:
+        data:
+          failed_items:
+            - id: '999'
+
+  - name: Try remove agent by name
+
+    request:
+      url: "{protocol:s}://{host:s}:{port:d}/agents/bad_id"
+      method: DELETE
+      headers:
+        Authorization: "Bearer {test_login_token}"
+
+    response:
+      status_code: 400
+      body:
+        <<: *error_spec
+
+  - name: Remove an agent
+
+    request:
+      url: "{protocol:s}://{host:s}:{port:d}/agents/006"
+      method: DELETE
+      headers:
+        Authorization: "Bearer {test_login_token}"
+      params:
+        purge: True
+
+    response:
+      status_code: 200
+      body:
+        data:
+          affected_items: ["006"]
+        message: !anystr
+
+---
+test_name: DELETE /agents/:agent_id/group
+
+marks:
+  - usefixtures:
+      - agents_test
+
+stages:
+  - name: Try remove group from agent 000 (manager)
+
+    request:
+      url: "{protocol:s}://{host:s}:{port:d}/agents/000/group"
+      method: DELETE
+      headers:
+        Authorization: "Bearer {test_login_token}"
+
+    response:
+      status_code: 400
+      body: &error_response
+        code: !anyint
+        dapi_errors:
+          master-node:
```

Comment:
It is possible to use a variable for setting the node name? If you have a master node whose name is different to `master-node`, the test will fail.

Reply:
Im going to change the way we check the API error results, now compare only error code